src/app/garfile/resperson/post.controller.js

```diff
@@ -0,0 +1,67 @@
+const logger = require('../../../common/utils/logger')(__filename);
+const CookieModel = require('../../../common/models/Cookie.class');
+const fixedBasedOperatorOptions = require('../../../common/seeddata/fixed_based_operator_options.json');
+const utils = require('../../../common/utils/utils');
+const validator = require('../../../common/utils/validator');
+const validations = require('../../resperson/validations');
+const resPersonApi = require('../../../common/services/resPersonApi');
+const garApi = require('../../../common/services/garApi');
+
+module.exports = async (req, res) => {
+    const cookie = new CookieModel(req);
+    const { buttonClicked } = req.body;
+    const resPersonId = req.body.addResponsiblePerson;
+    if (resPersonId && buttonClicked === 'Add to GAR') {
+        try {
+            res.redirect('/garfile/resperson?resPersonId=' + resPersonId);
+        } catch (err) {
+            logger.error('API failed to retrieve gar responsible person');
+            logger.error(err);
+            res.render('app/garfile/resperson/index', context, {
+                errors: [{
+                    message: 'Problem retrieving GAR',
+                }],
+            });
+        };
+    } else {
+
+        const responsiblePerson = utils.getResponsiblePersonFromReq(req);
```

Comment:
`nit`: Mix of the `async` and promises.
* In the future, should choose one or the other (ideally `async`, but it is another choice.
* Changing this can break tests and is large, so I don't really want it now.